Automattic/wp-calypso
# Stats: Update processing of post titles on the post details page

Related to #

## Proposed Changes

Fixes a potential bug in the string processing and updates truncation logic to break on whitespace only. The end result is that we won't get the ellipsis mid-word.

### Before with mid-word break

<img width="735" alt="SCR-20241216-qbyu" src="https://github.com/user-attachments/assets/315aca9c-1fcc-4a35-95b6-3a2e49e50b98" />

### After when breaking on whitespace

<img width="734" alt="SCR-20241216-qccm" src="https://github.com/user-attachments/assets/60824f85-4bc4-48c3-bd4f-b4fb07751455" />

## Why are these changes being made?


A small improvement and potential fix that I stumbled on when investigating the following:

https://github.com/Automattic/red-team/issues/323

## Testing Instructions



* Spin up the Calsypo Live site.
* Visit the All Posts page.
* Choose a post with a long title and click on the "xyz recent views" link to jump to the Post details page.
* Confirm the title is truncated on a word boundary instead of mid-word.

## Pre-merge Checklist
